File: programmers/lv2/기출문제/요격 시스템.py
def solution(targets):
    answer = 0
    
    targets.sort(key=lambda x: (x[0], x[1]))
        
    end_point = targets[0][1]
    targets = targets[1:]
    end_point_candidate = 100000000
    for target in targets:
        end_point_candidate = min(end_point_candidate, target[1])
        if target[0] >= end_point or target[0] >= end_point_candidate:
            answer += 1
            end_point = target[1]
            end_point_candidate = 100000000
    
    return answer + 1

File: programmers/lv2/연습문제/요격 시스템.java
import java.util.*;

class Solution {
    public int solution(int[][] targets) {
        int answer = 0;
        
        Arrays.sort(targets, (a, b) -> {
            if(a[1] == b[1]) {
                return Integer.compare(a[0], b[0]);
            }
            return Integer.compare(a[1], b[1]);
        });
        
        int prev = 0;
        for(int[] target: targets) {
            if(target[0] >= prev) {
                prev = target[1];
                answer++;
            }
        }
        
        return answer;
    }
}
